code/modules/power/singularity/particle_accelerator/particle_accelerator.dm

```diff
@@ -206,35 +216,39 @@ So, hopefully this is helpful if any more icons are to be added/changed/wonderin
 
 	switch(src.construction_state)//TODO:Might be more interesting to have it need several parts rather than a single list of steps
 		if(0)
-			if(iswrench(O))
+			if(istype(O, /obj/item/weapon/wrench) && !isinspace())
 				playsound(src.loc, 'sound/items/Ratchet.ogg', 75, 1)
 				src.anchored = 1
 				user.visible_message("[user.name] secures the [src.name] to the floor.", \
 					"You secure the external bolts.")
 				temp_state++
 		if(1)
-			if(iswrench(O))
+			if(istype(O, /obj/item/weapon/wrench))
 				playsound(src.loc, 'sound/items/Ratchet.ogg', 75, 1)
 				src.anchored = 0
 				user.visible_message("[user.name] detaches the [src.name] from the floor.", \
 					"You remove the external bolts.")
 				temp_state--
-			else if(iscoil(O))
-				if(O:use(1,user))
+			else if(istype(O, /obj/item/stack/cable_coil))
+				var/obj/item/stack/cable_coil/C = O
+				if(C.use(1))
 					user.visible_message("[user.name] adds wires to the [src.name].", \
 						"You add some wires.")
 					temp_state++
+				else
+					user << "<span class='warning'>You need one length of cable to wire the [src.name]!</span>"
+					return
 		if(2)
-			if(iswirecutter(O))//TODO:Shock user if its on?
+			if(istype(O, /obj/item/weapon/wirecutters))//TODO:Shock user if its on?
 				user.visible_message("[user.name] removes some wires from the [src.name].", \
 					"You remove some wires.")
 				temp_state--
-			else if(isscrewdriver(O))
+			else if(istype(O, /obj/item/weapon/screwdriver))
 				user.visible_message("[user.name] closes the [src.name]'s access panel.", \
 					"You close the access panel.")
 				temp_state++
 		if(3)
-			if(isscrewdriver(O))
+			if(istype(O, /obj/item/weapon/screwdriver))

```

Comment:
why would you change these
we have those helper procs for specifically this reason, everything should be using them, not a direct istype